src/wallet/rpcwallet.cpp

```diff
@@ -2350,6 +2350,79 @@ static RPCHelpMan settxfee()
+        RPCExamples{
+            ""
+            "\nSet a fee rate of 1 " + CURRENCY_ATOM + "/vB\n"
+            + HelpExampleCli("setfeerate", "1") +
+            "\nSet a fee rate of 3.141 " + CURRENCY_ATOM + "/vB\n"
+            + HelpExampleCli("setfeerate", "3.141") +
+            "\nSet a fee rate of 7.75 " + CURRENCY_ATOM + "/vB with named arguments\n"
+            + HelpExampleCli("-named setfeerate", "amount=\"7.75\"") +
+            "\nSet a fee rate of 25 " + CURRENCY_ATOM + "/vB with the RPC\n"
+            + HelpExampleRpc("setfeerate", "25")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue {
+            std::shared_ptr<CWallet> const rpc_wallet{GetWalletForJSONRPCRequest(request)};
+            if (!rpc_wallet) return NullUniValue;
+            CWallet& wallet = *rpc_wallet;
+
+            LOCK(wallet.cs_wallet);
+
+            CFeeRate amount{CFeeRate::FromSatB(AmountFromValue(request.params[0]))};
+            std::string amount_str{amount.ToString(FeeEstimateMode::SAT_VB)};
+            std::string current_amount_str{wallet.m_pay_tx_fee.ToString(FeeEstimateMode::SAT_VB)};
+            CFeeRate relay_min_feerate{wallet.chain().relayMinFee().GetFeePerK()};
+            CFeeRate wallet_min_feerate{wallet.m_min_fee.GetFeePerK()};
+            CFeeRate wallet_max_feerate{CFeeRate::FromBtcKb(wallet.m_default_max_tx_fee)};
+            std::string result, error;
+
+            if (amount == CFeeRate{0}) {
+                wallet.m_pay_tx_fee = amount;
+                result = "Fee rate for transactions with this wallet successfully unset. By default, automatic fee selection will be used.";
+            } else if (amount < relay_min_feerate) {
+                error = strprintf("The requested fee rate of %s cannot be less than the minimum relay fee rate of %s. The current setting of %s for this wallet remains unchanged.", amount_str, relay_min_feerate.ToString(FeeEstimateMode::SAT_VB), current_amount_str);
+            } else if (amount < wallet_min_feerate) {
+                error = strprintf("The requested fee rate of %s cannot be less than the wallet min fee rate of %s. The current setting of %s for this wallet remains unchanged.", amount_str, wallet_min_feerate.ToString(FeeEstimateMode::SAT_VB), current_amount_str);
+            } else if (amount > wallet_max_feerate) {
+                error = strprintf("The requested fee rate of %s cannot be greater than the wallet max fee rate of %s. The current setting of %s for this wallet remains unchanged.", amount_str, wallet_max_feerate.ToString(FeeEstimateMode::SAT_VB), current_amount_str);
+            } else {
+                wallet.m_pay_tx_fee = amount;
+                result = "Fee rate for transactions with this wallet successfully set to " + amount_str;
+            }
+            CHECK_NONFATAL(result.empty() != error.empty());
+
+            UniValue obj{UniValue::VOBJ};
+            obj.pushKV("wallet_name", wallet.GetName());
```

Comment:
c6e9bfd1bd9d6a65673fbe1c278a7c7ada35273d

Why return wallet name? What about returning previous effective value?

Reply:
why would the user care about the previous value? If they did, they could call getwalletinfo

Reply:
Right, the same goes for the wallet name.

Reply:
I'd prefer to keep wallet name.

Reply:
No strong opinion about the wallet name in the response. I think it does not hurt and in asynchronous environments it might simplify the client code.

Reply:
I think we should establish a guideline first on what input parameters to return. I don't think we've done this in the past and this could lead to user confusion: "What am I supposed to do with the return parameter that is equal to the input parameter? Assert that they are the same? Ignore?"

If this is something useful to return for every wallet RPC, then all wallet RPCs should be updated.

Reply:
I see it as good for rpcs that make per-wallet changes, like here and upgradewallet as a recent example (see upgradewallet help). It would also be good in rpc/cli options that return info on a per-wallet basis, like getwalletinfo but also -rpcwallet= -getinfo, for example.